plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionFields.java

```diff
@@ -70,16 +70,59 @@ public static PartitionSpec parsePartitionFields(Schema schema, List<String> fie
         }
     }
 
-    public static void parsePartitionField(PartitionSpec.Builder builder, String field)
+    private static void parsePartitionFields(Schema schema, List<String> fields, PartitionSpec.Builder builder, String field)
     {
-        boolean matched = tryMatch(field, IDENTITY_PATTERN, match -> builder.identity(fromIdentifierToColumn(match.group()))) ||
-                tryMatch(field, YEAR_PATTERN, match -> builder.year(fromIdentifierToColumn(match.group(1)))) ||
-                tryMatch(field, MONTH_PATTERN, match -> builder.month(fromIdentifierToColumn(match.group(1)))) ||
-                tryMatch(field, DAY_PATTERN, match -> builder.day(fromIdentifierToColumn(match.group(1)))) ||
-                tryMatch(field, HOUR_PATTERN, match -> builder.hour(fromIdentifierToColumn(match.group(1)))) ||
-                tryMatch(field, BUCKET_PATTERN, match -> builder.bucket(fromIdentifierToColumn(match.group(1)), parseInt(match.group(2)))) ||
-                tryMatch(field, TRUNCATE_PATTERN, match -> builder.truncate(fromIdentifierToColumn(match.group(1)), parseInt(match.group(2)))) ||
-                tryMatch(field, VOID_PATTERN, match -> builder.alwaysNull(fromIdentifierToColumn(match.group(1))));
+        for (int i = 1; i < schema.columns().size() + fields.size(); i++) {
+            try {
+                parsePartitionField(builder, field, i == 1 ? "" : "_" + i);
+                return;
+            }
+            catch (IllegalArgumentException e) {
+                if (e.getMessage().contains("Cannot create partition from name that exists in schema")
```

Comment:
Did you by chance open a ticket on iceberg to have a more structured way of identifying such a problem?